datadog/resource_datadog_synthetics_test_.go

```diff
@@ -830,29 +838,50 @@ func isTargetOfTypeInt(assertionType datadogV1.SyntheticsAssertionType, assertio
 
 func buildSyntheticsTestStruct(d *schema.ResourceData) *datadogV1.SyntheticsTestDetails {
 	request := datadogV1.NewSyntheticsTestRequest()
-	if attr, ok := d.GetOk("request.method"); ok {
+	k := NewResourceDataKey(d, "")
+	parts := ""
+	if v, ok := k.GetOkWith("request"); ok && v != nil && len(v.(map[string]interface{})) != 0 {
+		parts = "request"
+	} else {
+		parts = "request_definition.0"
+	}
+	k.Add(parts)
+	if attr, ok := k.GetOkWith("method"); ok {
 		request.SetMethod(datadogV1.HTTPMethod(attr.(string)))
 	}
-	if attr, ok := d.GetOk("request.url"); ok {
+	if attr, ok := k.GetOkWith("url"); ok {
 		request.SetUrl(attr.(string))
 	}
-	if attr, ok := d.GetOk("request.body"); ok {
+	if attr, ok := k.GetOkWith("body"); ok {
 		request.SetBody(attr.(string))
 	}
-	if attr, ok := d.GetOk("request.timeout"); ok {
-		timeoutInt, _ := strconv.Atoi(attr.(string))
+	if attr, ok := k.GetOkWith("timeout"); ok {
+		var timeoutInt int
+		// first try to convert to int if we're getting from the new TypeList field
```

Comment:
Sanity checking; is this because everything in a TypeMap is a string, regardless of what the Type is set as?

Reply:
yes